crowbar_framework/app/models/api/node.rb

```diff
@@ -158,7 +152,7 @@ def upgrade
 
       # this is just a fallback check, we should know by checking the global status that the action
       # should not be executed on already upgraded node
-      return true if upgraded?
+      return true if @node.file_exists? "/var/lib/crowbar/upgrade/node-upgraded-ok"
```

Comment:
Along the same lines of my previous suggestion to calculate the path to these status files based on the script name, I think it would be nicer to do:

    return true if @node.step_completed? :admin_upgrade